cmd/rdiff/delta.go

```diff
@@ -1,8 +1,63 @@
 package main
 
 import (
+	"io/ioutil"
+	"os"
+
+	"github.com/balena-os/librsync-go"
+	"github.com/sirupsen/logrus"
 	"github.com/urfave/cli"
 )
 
+var (
+	magic            = librsync.BLAKE2_SIG_MAGIC
+	blockLen  uint32 = 512
+	strongLen uint32 = 32
+	bufSize          = 65536
+)
+
 func CommandDelta(c *cli.Context) {
+	if len(c.Args()) > 3 {
+		logrus.Warnf("%d additional arguments passed are ignored", len(c.Args())-2)
+	}
+
+	if c.Args().Get(0) == "" {
+		logrus.Fatalf("Missing basis file")
+	}
+
+	if c.Args().Get(1) == "" {
+		logrus.Fatalf("Missing delta file")
+	}
+	if c.Args().Get(2) == "" {
+		logrus.Fatalf("Missing newfile file")
+	}
```

Comment:
We should respect the order of command-line arguments that is documented here: https://github.com/balena-os/librsync-go/blob/master/cmd/rdiff/main.go#L45 .

The idea is to keep the output file as the last argument (for consistency with other commands and also to keep the same argument order as in the original `rdiff` tool written in C.)

Also, the first argument should be the signature, not the basis file. More on this on a comment below.

Reply:
I updated my PR - once tried it with rdiff (from librsync) and with the same parameter with balena/rsync.